src/v2/BaseLoginRouter.js

```diff
@@ -238,17 +239,20 @@ export default Router.extend({
     }
 
     // render Controller
-    this.unload();
-    const controllerOptions = _.extend({
-      el: this.el,
-      settings: this.settings,
-      appState: this.appState
-    }, options);
-    this.controller = new Controller(controllerOptions);
-
-    // Bubble up all controller events
-    this.listenTo(this.controller, 'all', this.trigger);
-
+    if (Controller) {
+      this.unload();
+
+      const controllerOptions = _.extend({
+        el: this.el,
+        settings: this.settings,
+        appState: this.appState
+      }, options);
+      this.controller = new Controller(controllerOptions);
+
+      // Bubble up all controller events
+      this.listenTo(this.controller, 'all', this.trigger);
+    }
+    
     this.controller.render();
```

Comment:
In what case do we not have `Controller` ? In that case, won't the next line
```
this.controller.render()
```
throw an exception? (no property render on undefined)

Reply:
`Controller` will be undefined only when `render` is called on `interactionCanceled` handler:
https://github.com/okta/okta-signin-widget/blob/d3b02d7a0a3de1befdd71eddd5948d9fc64d8c95/src/v2/BaseLoginRouter.js#L83
In this case `this.controller` should be already defined.

Reply:
It is because when you trigger `interactionCanceled` the `controller` context should be passed and then you can call render on that context?

Reply:
`Back to sign in` is basically a redirect to login page and also clears any stateToken stored in sessionStorage. Can't we do something similar for `interactionCodeFlow` ? like just delete any existing token and do a page refresh to restart the login flow

Reply:
Reverted changes to `render` method. Pass controller as context to `interactionCanceled` event